receiver/k8sobjectsreceiver/receiver_test.go

```diff
@@ -294,3 +296,62 @@ func TestExcludeDeletedTrue(t *testing.T) {
 
 	assert.NoError(t, r.Shutdown(ctx))
 }
+
+func TestReceiverWithLeaderElection(t *testing.T) {
+	fakeLeaderElection := &k8sleaderelectortest.FakeLeaderElection{}
+	fakeHost := &k8sleaderelectortest.FakeHost{
+		FakeLeaderElection: fakeLeaderElection,
+	}
+	leaderElectorID := component.MustNewID("k8s_leader_elector")
+
+	mockClient := newMockDynamicClient()
+	rCfg := createDefaultConfig().(*Config)
+	rCfg.makeDynamicClient = mockClient.getMockDynamicClient
+	rCfg.makeDiscoveryClient = getMockDiscoveryClient
+	rCfg.ErrorMode = PropagateError
+	rCfg.Objects = []*K8sObjectsConfig{
+		{
+			Name: "pods",
+			Mode: PullMode,
+		},
+	}
+	rCfg.K8sLeaderElector = &leaderElectorID
+
+	r, err := newReceiver(
+		receivertest.NewNopSettings(metadata.Type),
+		rCfg,
+		consumertest.NewNop(),
+	)
+	require.NoError(t, err)
+	kr := r.(*k8sobjectsreceiver)
+	sink := new(consumertest.LogsSink)
+	kr.consumer = sink
+
+	// Setup k8s resources.
+	numPods := 2
+	mockClient.createPods(
+		generatePod("pod1", "default", map[string]any{
+			"environment": "production",
+		}, "1"),
+		generatePod("pod2", "default", map[string]any{
+			"environment": "production",
+		}, "1"),
+	)
+
+	err = kr.Start(context.Background(), fakeHost)
+	require.NoError(t, err)
+
+	// elected leader
+	fakeLeaderElection.InvokeOnLeading()
+
+	expectedNumMetrics := numPods
+	var initialLogRecordCount int
+	require.Eventually(t, func() bool {
+		initialLogRecordCount = sink.LogRecordCount()
+		return initialLogRecordCount == expectedNumMetrics
+	}, 20*time.Second, 100*time.Millisecond,
+		"logs not collected")
+
+	// lost election
+	fakeLeaderElection.InvokeOnStopping()
```

Comment:
Would it make sense to test if it really isn't a leader anymore? E.g. no more logs are ingested?